File: frontend/src/types/stats.ts
export interface StatsOverview {
  today_query_count: number
  active_tokens: number
  active_users: number
  dictionary_count: number
}

export interface StatRow {
  id: number | null
  label: string
  query_count: number
  rate_limited_count: number
}

export interface TopWordRow {
  word: string
  count: number
}

export type StatsDimension = 'token' | 'user' | 'date' | 'source'

File: backend/app/schemas/stats.py
from pydantic import BaseModel


class OverviewOut(BaseModel):
    today_query_count: int
    active_tokens: int
    active_users: int
    dictionary_count: int


class StatRow(BaseModel):
    id: int | None
    label: str
    query_count: int
    rate_limited_count: int


class TopWordRow(BaseModel):
    word: str
    count: int
